architecture/openshift_model.adoc

```diff
@@ -45,10 +45,16 @@ The custom build strategy is very similar to *Docker build* strategy, but users
 == DeploymentConfig
 
 == Image
+OpenShift stores information about Docker images including the "pull spec" (what you'd use to pull the image) and complete metadata about the image (e.g. command, entrypoint, environment variables, etc.). Images in OpenShift are immutable.
```

Comment:
These explanations of OpenShift objects is fine as-is, but if it is easy to link to / include the JSON structure of each, I think that would be valuable. I tried to do the same for the k8s objects.